docroot/modules/custom/dcc_gtd_registration/src/Form/GlobalTrainingRegistrationForm.php

```diff
@@ -0,0 +1,393 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\dcc_gtd_registration\Form\GlobalTrainingRegistrationForm
+ */
+
+namespace Drupal\dcc_gtd_registration\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\PrependCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Entity\Entity;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\StatusMessages;
+use Drupal\country;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class GlobalTrainingRegistrationForm
+ * @package Drupal\dcc_gtd_registration\Form
+ */
+class GlobalTrainingRegistrationForm extends FormBase {
+  protected $entityTypeManager;
+
+  /**
+   * GlobalTrainingRegistrationForm constructor.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $containerInterface
+   * @return static
+   */
+  public static function create(ContainerInterface $containerInterface) {
+    return new static(
+      $containerInterface->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * @return string
```

Comment:
For the methods that are inherited or implements an interface please use inheritdoc.